components/MyLayout.js

```diff
@@ -1,32 +1,138 @@
 import Head from 'next/head'
+import React, { Component } from 'react'
+import { withAuth } from 'use-auth0-hooks'
 
+import VerifyEmailScreen from '../components/verify-email-screen'
+import { ADMIN_USER_PATH, API_USER_PATH, AUTH0_SCOPE } from '../util/constants'
+import { fetchUser as fetchUserRaw } from '../util/middleware'
+import { renderChildrenWithProps } from '../util/ui'
 import NavBar from './NavBar'
 
-export default ({ children }) => (
-  <div>
-    <Head>
-      <title>OTP Admin Dashboard</title>
-    </Head>
-    <NavBar />
-    <main>
-      <div className='container'>
-        {children}
+/**
+ * Fetches user preferences, or if none available, make an initial user preference object, and return the result.
+ */
+async function fetchUser (route, proc, auth) {
+  const { accessToken } = auth
+
+  try {
+    const result = await fetchUserRaw(route, proc, accessToken)
+
+    // Beware! On AWS API gateway, if a user is not found in the middleware
+    // (e.g. they just created their Auth0 password but have not completed the account setup form yet),
+    // the call above will return, for example:
+    // {
+    //    status: 'success',
+    //    data: {
+    //      "result": "ERR",
+    //      "message": "No user with id=000000 found.",
+    //      "code": 404,
+    //      "detail": null
+    //    }
+    // }
+    //
+    // The same call to a middleware instance that is not behind an API gateway
+    // will return:
+    // {
+    //    status: 'error',
+    //    message: 'Error get-ing user...'
+    // }
+    // TODO: Improve AWS response.
+
+    const resultData = result.data
+    const isNewAccount = result.status === 'error' || (resultData && resultData.result === 'ERR')
+
+    if (!isNewAccount) {
+      return resultData
+    } else {
+      return null
+    }
+  } catch (error) {
+    // TODO: improve error handling.
+    alert(`An error was encountered:\n${error}`)
+  }
+}
+
+class MyLayout extends Component {
+  constructor () {
+    super()
+
+    this.state = {
+      adminUser: null,
+      apiUser: null,
+      isUserFetched: false,
+      isUserRequested: false
+    }
+  }
+
+  async componentDidUpdate () {
```

Comment:
I feel like all this auth0 stuff shouldn't be in a file dedicated to layouts. A better place would probably be in the `pages/_app` file.

Reply:
Unfortunately that cannot be in `pages/_app.js` because the `<Auth0Provider>` component is there and provides the `auth` props to all the children. It can be in a component between `<Auth0Provider>` and `<Layout>`.

Reply:
@binh-dam-ibigroup, see my commits in the demo branch to reorganize some things. It contains an example of how I think some of this stuff should work.

Reply:
I defer to a discussion we may need to have (#10) regarding how nextjs recommends fetching data.